[MIG][18.0] account_invoice_import

alexis-via opened the pull request:
Few additionnal fixes/improvements (that I'll backport to v16):

- The 'Upload' button from the vendor bill tree view now uses account_invoice_import
- FIX import of invoices without partner match
- Improve method _cannot_refund_vat() to use field unece_type_code on account.tax
- Add message in chatter of updated partner from wizard
- Fix message in chatter of created partner from wizard
- Fix default expense account from product category

Requires https://github.com/OCA/edi/pull/1243

alexis-via committed:
Rename module from account_invoice_pdf_import to account_invoice_import

FIX raise error when we don't identify a partner

Update README

alexis-via committed:
Update FR translation

alexis-via committed:
Add instructions to use my PPA for poppler-utils on Ubuntu 14.04 LTS + instructions OCR

alexis-via committed:
Add support for direct import of XML files (YAML test provided)

Rename test/pdf/ to test/invoices/ to host XML invoices too

Commit:
Sometimes, it's easier to extract the total tax amount instead of the total untaxed (depends on invoice layout)

alexis-via committed:
Add an explaination about date parsing in invoice2data

Commit:
Add optional support for start/end dates

alexis-via committed:
Adapt to latest changes in invoice2data

PEP8 stuff

alexis-via committed:
Only change name_get of invoice for our use-case, not everywhere

alexis-via committed:
Update README with latest changes in invoice2data lib

Commit:
Add support for partner_email to match partners

PEP8 fix

alexis-via committed:
Update installation procedure.

alexis-via committed:
Add modules base_zugferd, account_invoice_import_zugferd and account_invoice_zugferd (see README.rst for more info)

Move the ZUGFeRD/CII-specific code of account_invoice_import to account_invoice_import_zugferd

alexis-via committed:
Log raw result of invoice2data (before we make modifications)

Disable test of invoice2data, because this lib is difficult to install

alexis-via committed:
Fix error in README spotted by Roel Adriaans

alexis-via committed:
Add module account_invoice_import_ubl

Add support for UBL refunds

If you have an UBL refund that I could use to test, please contact me !

alexis-via committed:
Use base_vat_sanitized

Convert YAML test to unittest

alexis-via committed:
Add support for import of e-fff invoices (Belgian standard)

alexis-via committed:
Add support for local invoice template for invoice2data, see README

alexis-via committed:
Small usability improvement

alexis-via committed:
Checks that the directory really exists

alexis-via committed:
Extract regular PDF invoice import from account_invoice_import to a dedicated module account_invoice_import_invoice2data

Update README.rst and headers to latest OCA conventions.

alexis-via committed:
Adapt code to plug the module 'ovh_supplier_invoice'

Better key names in the parsed_inv dict
parsed_inv doesn't need to be JSON serializable anymore (small drawback: the invoice is parsed a second time on the second step... but the second step is rarely used)

alexis-via committed:
Fix product matching

Move code from account_invoice_import_invoice2data to account_invoice_import

alexis-via committed:
FIX PEP8 and account_invoice_import_invoice2data unittests

Update REAME and some interface strings about UBL being an ISO standard
Small code changes

alexis-via committed:
Add module base_business_document_import and start to move some code to it.

alexis-via committed:
Better/cleaner pivot invoice dict format (no more IDs in the pivot invoice dict, cleaner organisation)

Code refactoring: move code in base_business_document_import, factorise code for tax matching (it was duplicated in UBL and ZUGFeRD)
Now support PDF with embedded UBL XML file
Enable unittests on account_invoice_import_ubl
More absolute xpath in account_invoice_import_ubl instead of relative xpath

WARNING: these are big changes, I may have broken a few details

alexis-via committed:
Add module base_ubl (common methods to generate and parse UBL files)

Adapt module account_invoice_import_ubl to use the new base_ubl module
Small fixes

alexis-via committed:
Add module base_phone_business_document_import

Add unitests in base_business_document_import
Small code enhancements/simplifications

alexis-via committed:
Add a unittest on _match_uom

Code cleanup/minor changes

alexis-via committed:
Add module base_business_document_import_stock

Rename key 'quantity' to 'qty' in all parsing dicts['lines']
Add common methods compare_lines() and post_create_or_update()
Make sure price_unit is always untaxed in UBL XML files

alexis-via committed:
Update invoice lines on "Update Existing" invoice import

Add button "Update from File" on supplier invoice form
Use float_round() instead of round()

alexis-via committed:
Fix crash

alexis-via committed:
Add multi-company ir.rule on account.invoice.import.config

alexis-via committed:
FIX Don't use _inherit = ['business.document.import']

because we want to have access to the code of the modules that inherit business.document.import

alexis-via committed:
Fix method call

Commit:
[IMP] handle the case where the xml file is generated with mime type
'text/xml'

alexis-via committed:
Add support for extraction + matching on website

PEP8 fix

alexis-via committed:
FIX two stupid bugs

Commit:
8.0 Add support for partner bank matching on invoice update (#6)

Add support for partner bank matching on invoice update (before, it was only supported on invoice creation)

Commit:
8.0 Use try/except when importing external libs (#8)

[FIX] LINT

Use try/except when importing external libs
Remove self.ensure_one() that has nothing to do in an api.model method

hbrunn committed:
[MIG] Make modules uninstallable

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
[ADD] migration of account_invoice_import_invoice2data and dependencies

alexis-via committed:
Add support for 'origin' field in invoice import

hbrunn committed:
[FIX] product.product#ean13 was renamed to barcode

hbrunn committed:
[FIX] tests

alexis-via committed:
Port base_business_document_import* to v10

Also port all the modules that generate the XML documents: account_invoice_ubl, account_invoice_zugferd, purchase_order_ubl and sale_order_ubl

Add active buttons on invoice import config

Note: it requires this fix on odoo 10: https://github.com/odoo/odoo/pull/15649

Port account_invoice_import_factur-x and account_invoice_import_ubl to v10.0

base_business_document_import: Add support for the creation of res.bank

Port purchase_order_import* to v10.0

Add ubl invoice generation option in accounting config page

Update to OCA coding standards

alexis-via committed:
Allow importing invoice without invoice_import_config on partner (usefull for third-party modules that use account_invoice_import)

Code cleanup and optimisations

alexis-via committed:
Several fixes in account_invoice_import

Reorganise menu entries for invoice import and update labels

alexis-via committed:
Improve XML for Factur-X

Make country_id field required on res.partner form view

alexis-via committed:
Add first support for the import of Factur-X invoices

Add support for start/end dates on lines
Better handling of invalid IBANs

alexis-via committed:
Add support for allowance and charges on Factur-X invoice import

To avoid headaches with invalid files, now check XSD of XML invoices before parsing it upon import
Move Factur-X/zugferd xsd to base_zugferd

alexis-via committed:
Fix bug #16 : add support for adjustment lines (per line and global)

Add support for Tax Due Date on invoice import

alexis-via committed:
account_invoice_factur-x: now generate refunds with document type = 381 instead of document type = 380 with negative amounts

Improve the code of refund import to support both negative invoices and positive refunds

alexis-via committed:
Pre-process parsed_inv even when calling directly _create_invoice()

alexis-via committed:
Add support for importing customer invoices (bug #24)

Add basic unit-tests in account_invoice_import

alexis-via committed:
Improve + modularize generation of UBL (up-port from v8 PR)

Add script mass_invoice_import.py
Code cleanup

alexis-via committed:
Finalise the use of the FacturX lib

One of the zugferd-specific code has been standardised in UNECE, so moved to the account_tax_unece module

alexis-via committed:
account_invoice_import: workaround a framework bug

Fix update invoice via import wizard

alexis-via committed:
account_invoice_import: Add partner_type to _match_partner method for better support out customer invoice import

Fix test suite of account_invoice_import (Agrolait is only a customer in v10)

alexis-via committed:
Partner <-> Invoice import config : switch M2O <-> O2M

alexis-via committed:
Add migration script

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0] Update to work with latest version of invoice2data (#51)

* Update to work with latest version of invoice2data

* Add requirements.txt file

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0] Several fixes + import invoices with VAT in a company that cannot deduct VAT (#64)

* Move code to a new dedicated method
* FIX update of existing draft invoice
  Improve error message of partner matching
  Move hook of partner matching to put it after VAT matching, which is the
  most reliable
* Add support for importing an invoice with VAT in a company that cannot deduc VAT
  Display nice error when account is missing on product
* Improve compliancy with Factur-X standard (using schematron)

oca-travis committed:
[UPD] Update account_invoice_import.pot

alexis-via committed:
Fix import of sale order with price_source='pricelist' when pricelist has visible discounts

Code improvements in sale_order_import
Add unit tests in sale_order_import
Use display_name instead of name_get()[0][1]

alexis-via committed:
Add support for multiple invoice import configs in invoice import wizard

Update other modules accordingly

alexis-via committed:
Add O2M field from import config to download config

Code update and cleanup

Commit:
[FIX] include some of the fixes proposed in #64

alexis-via committed:
Finalize the development account_invoice_download

Update the account_invoice_download_weboob following the changes in account_invoice_download
Add README.rst for the 2 modules

alexis-via committed:
Add support for additional module-specific weboob config parameters

Add weboob module management
Improve log handling (explicit error when wrong pwd)
account_invoice_import: by default, set tax to default tax of account when invoice_line_method = 1line_no_product

alexis-via committed:
account_invoice_import: add a mail gateway

Update text displayed in the invoice import wizard and make list of
supported formats modular (like in bank statement import)

alexis-via committed:
Add ability to run cron invoice download from admin for all companies in multi-company environnement

Fix my previous commit on sequence field on invoice import config (caused a crash when creating a new record)
Usability improvements on download config

alexis-via committed:
Add multi-company support in the invoice import mail gateway

alexis-via committed:
Check the VAT number of the destination partner, to make sure the business document is imported in the right company

oca-travis committed:
[UPD] Update account_invoice_import.pot

Commit:
[11.0][MIG] account_invoice_import

Commit:
[FIX] clean rebase  from V10 that where wrong ...

Commit:
[FIX] utf8 comment in python3

astirpe committed:
Fix base64 encode/decode + remove supplier_invoice_number + code review

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

OCA-git-bot committed:
[ADD] icon.png

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-11.0/edi-11.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-11-0/edi-11-0-account_invoice_import/

OCA-git-bot committed:
[UPD] README.rst

astirpe committed:
[12.0][MIG] account_invoice_import

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/

alexis-via committed:
account_invoice_download: start OCA import bill wizard when using the "Import Bill" button on journal dashboard

tv-openbig committed:
Added translation using Weblate (German)

tv-openbig committed:
Translated using Weblate (German)

Currently translated at 92.4% (97 of 105 strings)

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/de/

alexis-via committed:
[MIG] account_invoice_import_facturx to v12

Update technical name of module to remove '-'
Up-port PR #78 Auto-creation of bank accounts upon invoice import is now optional
Fix returned action when using the import wizard

oca-travis committed:
[UPD] Update account_invoice_import.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/

marylla committed:
Translated using Weblate (German)

Currently translated at 91.5% (97 of 106 strings)

Translation: edi-12.0/edi-12.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-12-0/edi-12-0-account_invoice_import/de/

yvaucher committed:
[IMP] account_invoice_import: black, isort, prettier

yvaucher committed:
[MIG] account_invoice_import: Migration to 13.0

yvaucher committed:
[IMP] account_invoice_import: parsed invoice hook

Add a hook to inject parsed invoice from othe means than a file

yvaucher committed:
[IMP] account_invoice_import: Missing partner hook

Add a hook for an action on missing partner

yvaucher committed:
Add a quick import invoice button

A button is added on the Vendor Bill form to import invoices in serie
This button can be accessed through the keyboard shortcut with Alt + M (or an other letter depending on the modules installed on the server

yvaucher committed:
Restore dashboard button overwrite

NL66278 committed:
[FIX] account_invoice_import: fix headers

NL66278 committed:
[FIX] account_invoice_import: Reduce method complexity.

NL66278 committed:
[FIX] account_invoice_import. Do not shadow config module with variable.

vrenaville committed:
[RM] invoice line doesn't exist on V13

alexis-via committed:
[MIG] account_invoice_import to v14

Add support for partner creation

StefanRijnhart committed:
[FIX] account_invoice_import: mail gateway exception

alexis-via committed:
account_invoice_import: Add USAGE.rst

oca-travis committed:
[UPD] Update account_invoice_import.pot

OCA-git-bot committed:
[UPD] README.rst

alexis-via committed:
account_invoice_import: fix analytic account field name

alexis-via committed:
account_invoice_import: FIX import config creation from partner

When you import an invoice via the wizard for a new partner that doesn't
exist yet, the wizard will propose you to create a new one. Once the
partner is saved, when you click on "0 invoice import configs", you
would get a bad value in the partner_id field. This commit fixes it.

alexis-via committed:
invoice_import: add button box div in account.invoice.import.config form view

This will be used by other modules such as account_invoice_download

OCA-git-bot committed:
account_invoice_import 14.0.1.1.0

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 94.3% (100 of 106 strings)

Translation: edi-14.0/edi-14.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import/fr/

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 96.4% (136 of 141 strings)

Translation: edi-14.0/edi-14.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import/nl/

sebalix committed:
[FIX] Fix access right issues when using actions

OCA-git-bot committed:
account_invoice_import 14.0.1.1.1

alexis-via committed:
[FIX] access to form view of partners for users who are not accountants

OCA-git-bot committed:
account_invoice_import 14.0.2.0.0

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 96.4% (136 of 141 strings)

Translation: edi-14.0/edi-14.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import/nl/

alexis-via committed:
account_invoice_import: Set a specific journal

You can now set a specific journal, either via the
account.invoice.import.config or by the pivot dict.
Fix setting invoice_date_due via pivot dict.
Improve multi-company handling in tests.
Add test for setting journal via pivot dict.
Add more checks in test.

simahawk committed:
account_invoice_import: tests use setUpClass

simahawk committed:
account_invoice_import: fix error logging

* do not spoil emails
* message_id is enough to find the msg in Odoo
* no need to be so verbose on where it happened since the logger name is precise

simahawk committed:
account_invoice_import: more tests for mail gateway

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_invoice_import 14.0.2.0.1

alexis-via committed:
[FIX] account_invoice_import: force tax amount on supplier refunds

Commit:
[UPD] Update account_invoice_import.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_invoice_import 14.0.2.1.0

alexis-via committed:
account_invoice_import: improve handling of simple PDF invoices

Implement what is described in issue #610:
- account_invoice_import_simple_pdf and account_invoice_import_invoice2data can now be use
d together
- when importing an invoice for an unknown partner, it creates an empty invoice with PDF a
s attachment
- when importing an invoice via mail gateway and the from email is set
on a partner that doesn't have any import config, create a supplier
invoice with PDF as attachment and with that partner
- when importing an invoice via mail gateway and the from email is set
on a partner that has a single-line import config, create a supplier
invoice with PDF as attachment, that partner and 1 invoice line with price_unit=0.

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import/

alexis-via committed:
account_invoice_import: skip mail attachments that are not PDF nor XML

FIX import of XML files via mail gateway
FIX attachment number in logs

klodr committed:
Translated using Weblate (French)

Currently translated at 94.3% (100 of 106 strings)

Translation: edi-14.0/edi-14.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import/fr/

Commit:
[UPD] Update account_invoice_import.pot

OCA-git-bot committed:
account_invoice_import 14.0.3.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-account_invoice_import
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-account_invoice_import/

alexis-via committed:
[MIG] account_invoice_import to v15

alexis-via committed:
[FIX] account_invoice_import: string remplacement in chatter and error messages

alexis-via committed:
[FIX] account_invoice_import: crash due to missing method

The method get_invoice_line_account() was removed from the "account"
module in odoo v13. So we now use the underlying method get_product_accounts()
Fixes bug #572

alexis-via committed:
account_invoice_import: update multi-string code to recommended way

alexis-via committed:
account_invoice_import: minor update of pivot format doc

alexis-via committed:
[MIG] account_invoice_import to v16

First draft migration, to have a first working version. Many more
changes are planned.

alexis-via committed:
[IMP] account_invoice_import: add support for discount

Add support for discount on lines in the multi-line create scenario
Fix force tax scenario and simplify the code

alexis-via committed:
[IMP] account_invoice_import: re-write module according to changes proposed in bug #803

- support multi-file in invoice import wizard
- move fields of account.invoice.import.config to res.partner
- remove field invoice_method (replace with invoice_import_single_line)
- remove the "Update" scenario

alexis-via committed:
[IMP] account_invoice_import: re-write partner create/update feature

Add import warnings in dedicated banner
Rename field invoice_import_warnings to import_warnings on account.move

alexis-via committed:
[IMP] account_invoice_import: set company_dependent=True on all partner fields

alexis-via committed:
[IMP] account_invoice_import: add migration script

alexis-via committed:
[FIX] account_invoice_import: bug in adjustment line creation

alexis-via committed:
[IMP] account_invoice_import: fine-tune code

Implement/update several TODOs in the code
Add warning banner in the imported invoice.
Avoid a crash when odoo cannot force total VAT amount when there are not
taxes in the invoice.
Add notification pop-up upon Vendor bill import wizard validation
Improve usability of the create or update partner wizard

alexis-via committed:
[IMP] account_invoice_import: improve string

Update fr translation

alexis-via committed:
[FIX] base_business_document_import: avoid double entries in chatter

Fix crash in _match_taxes() when _match_tax returne None
Add raise_exception argument in several methods

alexis-via committed:
[FIX] account_invoice_import: pre-commit fix

Commit:
[UPD] Update account_invoice_import.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

OCA-git-bot committed:
[BOT] post-merge updates

alexis-via committed:
[IMP] pre-commit auto fixes/updates

alexis-via committed:
[MIG] account_invoice_import: migrate 16 -> 18

The 'Upload' button from the vendor bill tree view now uses account_invoice_import
FIX import of invoices without partner match
Improve method _cannot_refund_vat() to use field unece_type_code on account.tax
Add message in chatter of updated partner from wizard
Fix message in chatter of created partner from wizard
Fix default expense account from product category

alexis-via committed:
[IMP] account_invoice_import: restore native 'Upload' widget on accounting dashboard

alexis-via committed:
[IMP] account_invoice_import: improve code that update an existing partner

hbrunn reviewed (approved):
please rebase and squelch the [warning](https://github.com/OCA/edi/actions/runs/19970586371/job/57274030494?pr=1244#step:8:376) though

alexis-via committed:
[FIX] account_invoice_import: fix test of mail gateway in multi-company setup

Remove test 'test_email_gateway' that always give a logger.error message
because the destination email to not configured on any company

alexis-via commented:
/ocabot merge nobump

hbrunn commented on account_invoice_import/tests/test_invoice_import.py:
@mute_logger should have fixed that too I think?
